src/test/java/seedu/cafectrl/command/ListIngredientCommandTest.java

```diff
@@ -17,33 +19,44 @@ class ListIngredientCommandTest {
     public void execute_validIndex_printsIngredients() {
         ArrayList<Dish> menuItems = new ArrayList<>();
         menuItems.add(new Dish("Chicken Rice",
-                new ArrayList<>(Arrays.asList(new Ingredient("Rice", 1, "cup"),
-                        new Ingredient("Chicken", 100, "g"))), 8.0F));
+                new ArrayList<>(Arrays.asList(new Ingredient("Rice", 100, "g"),
+                        new Ingredient("Chicken", 200, "g"))), 8.0F));
         menuItems.add(new Dish("Chicken Sandwich",
                 new ArrayList<>(Arrays.asList(new Ingredient("Lettuce", 100, "g"),
                         new Ingredient("Chicken", 50, "g"))), 5.0F));
         Menu menu = new Menu(menuItems);
 
-        ArrayList<String> actualOutput = new ArrayList<>();
-        Ui ui = new Ui() {
-            @Override
-            public void showToUser(String... message) {
-                actualOutput.addAll(Arrays.asList(message));
-            }
-        };
+        Ui ui = new Ui();
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        PrintStream consoleStream = new PrintStream(baos);
+
+        PrintStream originalOut = System.out;
+        System.setOut(consoleStream);
 
         int indexToSelect = 1;
         ListIngredientCommand listIngredientCommand = new ListIngredientCommand(indexToSelect, menu, ui);
         listIngredientCommand.execute();
 
-        String expectedOutput = "Chicken Rice Ingredients: \n"
-                + "Rice - 1cup\n"
-                + "Chicken - 100g\n";
+        String actualOutput = baos.toString().trim();
+        System.setOut(originalOut);
+
+        String expectedOutput = "+-------------------------------------------------------+"
```

Comment:
Perhaps you should use the Strings stored in Messages.java instead of "magic strings"

Reply:
Initially, I felt like it would be easier to visualise the expected output for the junit test if it was displayed as such but I agree with what you brought up. Thank you!